analytics.go

```diff
@@ -239,10 +243,28 @@ func (c *Client) Close() error {
 	return nil
 }
 
+func (c *Client) sendAsync(msgs []interface{}) {
+	c.inflight++
+	go func() {
+		c.sendDone <- c.send(msgs)
+	}()
+}
+
+func (c *Client) awaitInflight() {
+	for c.inflight > 0 {
+		c.verbose("waiting for %d requests to finish..", c.inflight)
+		err := <-c.sendDone
+		if err != nil {
+			c.logf(err.Error())
+		}
+		c.inflight--
+	}
+}
```

Comment:
This really feels like you're reinventing the wheel, why not simply use a `sync.WaitGroup` object instead? The only benefit I'm seeing is with the code you submitted you get to know how many requests are pending, it doesn't bring a lot of value (it's only ever shown in the logs in verbose mode).

I feel like the right approach is to use a standard library construct, until we have a problem to solve that really needs us to know how many requests are pending (even then we could use a pair of a sync.WaitGroup and an atomic counter to get the information).